ten-simple-rules-dockerfiles.Rmd

```diff
@@ -124,29 +124,29 @@ In this case, proceed with following the remaining rules to write your `Dockerfi
 Automated builds can be complex to set up, and details are out of scope of this article.
-The documentation and tools of container registries, e.g., [Docker Hub](https://docs.docker.com/docker-hub/builds/) or [GitLab](https://docs.gitlab.com/ee/user/packages/container_registry/index.html#build-and-push-images) instances, but also of CI platforms, e.g. [GitHub actions](https://github.com/actions/starter-workflows/tree/master/ci), a GitHub actions for `repo2docker` [@husain_repo2docker-action_2019; @scottyhq_repo2docker-githubci_2019], or [CircleCI](https://circleci.com/orbs/registry/orb/circleci/docker#commands-build), can help you get started.
+The documentation for, and toolsets provided by  container registries such as [Docker Hub](https://docs.docker.com/docker-hub/builds/) or [GitLab](https://docs.gitlab.com/ee/user/packages/container_registry/index.html#build-and-push-images) as well as CI platforms such as [GitHub actions](https://github.com/actions/starter-workflows/tree/master/ci) or [CircleCI](https://circleci.com/orbs/registry/orb/circleci/docker#commands-build) can help you get started; tools may also be available to automate builds using intermediate tools, such as the `repo2docker` Github action [@husain_repo2docker-action_2019; @scottyhq_repo2docker-githubci_2019],
 You should avoid publishing an image in a public registry with `docker push`, because it is opaquely breaking the linkage between `Dockerfile` and image.
 
 A tag like `latest` is good in that security fixes and updates are likely present, however it is bad in that it's a moving target so that it is more likely that an updated base could break your workflow.
 Other tags that should be avoided are `dev`, `devel`, or `nightly` that may provide possibly unstable versions of the software.
-As an example, the Python container on Docker Hub will be released with new versions of Python.
+Official builds are often tagged with corresponding release version numbers as well as with the `latest` tag.
 If you build a container with `python:latest` that only supports Python version 3, when the base is updated to Python 4 it's likely that your software won't work as expected. 
 In this case, a tag like `python:3.5` might be a good choice, where the third component of the version, i.e., `3.5.x`, ensures that security patches and bugfixes that won't break your code, i.e., changes that are backwards compatible, are applied (cf. semantic versioning, [@preston-werner_semantic_2013]).
+
+Containers can grow quickly in size, particularly when data is involved, and you should keep this in mind. Smaller images are often indicated by having `slim` or `minimal` as part of the tag.  If you need to build smaller containers, consider a `busybox` base or a multi-stage build, which allows you to selectively copy files from one build step to another. Also take into account the libraries that you actually need.
+
```

Comment:
Did you copy paste this from somewhere? I'm pretty sure that I wrote that.

Reply:
Oup, now I see it below! I'm wondering why you moved it up? The previous paragraph here is talking about container tags, and then the next is talking about a base image, so it's awkward to suddenly bring up size in the middle. My preference is for where the text originally was - after talking about tags, base images, it flows more nicely into talking about size. The last line about dependencies also transitions nicely into the last paragraph.

Reply:
I was linking the use of tags, in particular the idea that as well as tagging a release number, a tag may also be flagging the weight of the build.

We could tweak the next para to mention something about base images and how we could choose lite ones over heavier ones?

Maybe s/thing like:

```
Container tags may also provide an indication as the size of a container.  Smaller images are often indicated by having `slim` or `minimal` as part of the tag. 

The container size may also be influenced by the base image. If you need to build smaller containers, consider a `busybox` base or a multi-stage build, which allows you to selectively copy files from one build step to another. Also take into account the libraries that you actually need.

More generally, when you choose a base image, try choose one with a Linux distribution that supports the software stack you are using, and also take into account the bases that are widely used by your community.
```

Reply:
It's less about tags, and more about container size. It just happens to be the case that _sometimes_ a tag like minimal or slim can hint at a smaller container, but doesn't guarantee. Since it's not explicitly about tags but rather size, this is why it transitions nicely after tags, but doesn't make sense to plop in the middle of the discussion.

Reply:
okay.. reordered back...

Reply:
Great, thank you!